server/modules/activityLog/repository/activityLogRepository.js

```diff
@@ -40,13 +40,32 @@ export const fetch = async (surveyId, activityTypes = null, offset = 0, limit =
           ${getSurveyDBSchema(surveyId)}.activity_log a
         WHERE
           system = false
-        ${activityTypes ? 'AND a.type IN ($1:csv)' : ''}
+        ${activityTypes ? 'AND a.type IN ($2:csv)' : ''}
     
       )
     SELECT
       l.*,
       u.name AS user_name,
-      r.uuid AS record_uuid
+      r.uuid AS record_uuid,
+      -- user activities keys
+      user_target.name AS target_user_name,
+      user_target.email AS target_user_email,
+      -- check if target user has been removed (not in auth_group_user table)
+      NOT EXISTS (    
+        SELECT * 
+        FROM  
+          public.survey s
+        JOIN
+          public.auth_group_user agu
+        ON
+          agu.user_uuid = user_target.uuid 
+        JOIN  
+          public.auth_group ag
+        ON 
+          ag.uuid = agu.group_uuid AND ag.survey_uuid = s.uuid
+        WHERE 
+          s.id = $1
+      ) as target_user_removed
```

Comment:
would selecting the target_user_uuid with left outer joins be more efficient?

Reply:
yes, I will change it